department-of-veterans-affairs/caseflow
# Justin/caseflow 4594 rpo mark task in progress

Resolves [CASEFLOW-4594](https://vajira.max.gov/browse/CASEFLOW-4594)

### Description
- Added a task action in the Education Assess Documentation Task to call the task action repo.
- Assessed reusability of and subsequently utilized preexisting vha functionality for marking tasks in progress
- Added method in the task action repository to mark Education Assess Documentation Tasks \(rpo tasks\) as in progress 
- Updated TASK_ACTIONS.json to include RPO\_MARK\_TASK\_IN\_PROGRESS
- Updated COPY.json and Task Action Repo to change "_VHA_ IN PROGRESS" labels to _ORGANIZATION_
- Updated pre_docket_spec to match new ORGANIZATION task labels
- Updated education assess documentation task spec file for marking rpo tasks in progress
- Created new feature test in the pre_docket_spec to verify tasks are correctly marked "in progress" in the application

### Acceptance Criteria
- [x] "Mark task in progress" task action is created
- [x] "Mark task in progress" modal is created
- [x] Upon success, show the appropriate success banner
- [x] Upon fail, show a failure (most likely a Slim Error Alert in the modal)
- [x] After confirmation the RPO task should have a status of "in_progress"

### Testing Plan
1. Log in as intake user to intake a new appeal. Add education issues and mark this appeal for predocketing.
2. Log in as EMO user and open queue. Set the view to the organization queue
3. Choose newly intaken appeal and send the appeal to a Regional Processing Office
4. Log in as an RPO user and open queue. Set the view to the same office the appeal was sent to by EMO. 
5. Choose the appeal from the Assigned tab, and click on the Actions dropdown.
6. Choose "Mark Task as In Progress", and accept the prompt.
7. A success banner will show on the next screen, and the appeal should now be in the "In Progress" tab


### User Facing Changes
![image](https://user-images.githubusercontent.com/95875751/167225384-b9d8face-507d-483e-a544-bf319239b65a.png)
![image](https://user-images.githubusercontent.com/95875751/167225401-758f49a4-f83d-41cd-8ee6-fdc8bd40d98e.png)
![image](https://user-images.githubusercontent.com/95875751/167225419-d5d77d3a-48de-4aab-a6ec-184c0373fdb7.png)
![image](https://user-images.githubusercontent.com/95875751/167225452-4d941655-9c08-4fca-9796-69896e28366f.png)
![image](https://user-images.githubusercontent.com/95875751/167225530-97c7d798-ee12-4e2a-b124-29331b531c90.png)


### Code Documentation Updates

### [Link to Designs](https://bahdigital.invisionapp.com/share/QMIAIA7TANS#/screens/296070391?browse)


### Code Changes (set against feature)
![github com_department-of-veterans-affairs_caseflow_compare_feature_CASEFLOW-2738 Justin_CASEFLOW-4594-rpo-mark-task-in-progress (1)](https://user-images.githubusercontent.com/95875751/167173180-725642bb-b7d6-4340-a686-4ca2184e5b10.png)
